waltz-server/src/main/java/com/wepay/waltz/server/WaltzServer.java

```diff
@@ -481,4 +483,16 @@ private static void logVersion() {
 
         logger.info("STARTING [name=" + title + " version=" + version + "]");
     }
+
+    private static void checkConfig(WaltzServerConfig config) {
+        int minFetchSize = (int) config.get(WaltzServerConfig.MIN_FETCH_SIZE);
+        int realtimeThreshold = (int) config.get(WaltzServerConfig.REALTIME_THRESHOLD);
+
+        // minFetchSize must be smaller than realtimeThreshold. Otherwise, the catchup feed task may be stalled.
+        // The catch-up feed task uses FIFO queue to manage feed contexts. It assumes there is always a transaction to send.
+        if (minFetchSize >= realtimeThreshold) {
+            throw new ConfigException("\"server.minFetchSize\" must be smaller than \"server.realtimeThreshold\"");
```

Comment:
Nit: should use WaltzServerConfig static variables instead of hard-coded names in ConfigException message.